components/keyboard_shortcuts/keyboard_shortcuts.ts

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {MessageDescriptor} from 'react-intl';
+
+import {t} from '../../utils/i18n';
+
+export type KeyboardShortcutsSeq = MessageDescriptor | {default: MessageDescriptor; mac?: MessageDescriptor};
+
+export function isMessageDescriptor(descriptor: KeyboardShortcutsSeq): descriptor is MessageDescriptor {
+    return Boolean((descriptor as MessageDescriptor).id);
+}
+
+export const KEYBOARD_SHORTCUTS = {
```

Comment:
IMO it would be better to create a index file for the keyboard_shortcuts and export members and defaults from one location ... that would make imports in other files more clear